real tensor prop for composite ops

pianpwk opened the pull request:
Fixes #135632

Adds real tensor propagation for decompositions, checking any symbols on their outputs

ezyang commented:
This doesn't actually have anything to do with composites, it has to do with ops that return the unbacked symbol as an inner size/stride

pianpwk committed:
real tensor prop for composite ops (#135717)

Summary:
Fixes https://github.com/pytorch/pytorch/issues/135632

Adds real tensor propagation for decompositions, checking any symbols on their outputs


Reviewed By: ezyang

Differential Revision: D62512551

Pulled By: pianpwk

pianpwk commented on torch/_subclasses/fake_tensor.py:
```diff
@@ -1898,17 +1899,33 @@ def maybe_to_real_tensor(t: T) -> Optional[Union[T, Tensor]]:
+                    go,
+                    tuple(pytree.tree_flatten(fake_out)),
+                    tuple(pytree.tree_flatten(real_out)),
+                )
```
Ah, just updated to use the old version by default. Hopefully that's better.
